plugins/language_jsx.lua

```diff
@@ -4,7 +4,7 @@ local syntax = require "core.syntax"
 
 syntax.add {
   name = "JSX",
-  files = { "%.jsx$" },
+  files = { "%.jsx$", "%.astro$" },
   comment = "//",
   patterns = {
     { pattern = "//.-\n",               type = "comment"  },
```

Comment:
I realize that this isn't really part of the PR, but could you also change this?
```suggestion
    { pattern = "//.*",                 type = "comment"  },
```